### ISSUE TITLE
"'errorValue' int 5 | Timed out" on successive calls to daemons()

### ISSUE THREAD

**Issue by wlandau** [shikokuchuo/mirai#48]
Re https://github.com/wlandau/crew/issues/61 and https://github.com/shikokuchuo/mirai/issues/47, I am able to reproduce some of my difficulties using just `mirai`. The following result happens both in RStudio and the terminal on my Macbook.

```r
library(mirai)
library(nanonext)
daemons(n = 1L, url = "ws://127.0.0.1:5000", dispatcher = TRUE, token = TRUE)
count <- 0L
while (TRUE) {
  count <- count + 1L
  out <- daemons()$daemons
  if (!is.matrix(out)) {
    break;
  }
  msleep(10) # or msleep(1) or msleep(100)
}
count
#> [1] 516
out
#> 'errorValue' int 5 | Timed out
```

At this point, my results vary. Sometimes the dispatcher exits when I call `daemons(n = 0L)`, sometimes it keeps running. Sometimes when I restart my R session in RStudio, the R console freezes or I get a crash. Other times, the restart succeeds and RStudio returns control to me at the R console.

**Comment by wlandau** [comment]
Update: I tried the above example with `msleep(1)`, `msleep(10)`, and `msleep(100)`, and they all lead to the same `'errorValue' int 5 | Timed out` at about `count == 525`. Interestingly, when `'errorValue' int 5 | Timed out` does appear, the dispatcher persists on my Macbook and exits on my Ubuntu setup.

**Comment by shikokuchuo** [comment]
It goes to just under 4,000 on my Ubuntu laptop. However I have narrowed this down...

**Comment by shikokuchuo** [comment]
This now works for me - it doesn't break, and I got to > 50,000 when I manually interrupted.

**Comment by wlandau** [comment]
Thank you so much! This is a definite improvement. Working on my Ubuntu machine, I do not see the same crashes I did before, and in order to continue to reproduce `'errorValue' int 5 | Timed out`, I had to move to `msleep(1)` or remove `msleep()` entirely. The last couple attempts at the following reached `count == 49454` and `count == 695220` on my Ubuntu machine at the time `out` reached `'errorValue' int 5 | Timed out`.

```r
library(mirai)
library(nanonext)
daemons(n = 1L, url = "ws://127.0.0.1:5000", dispatcher = TRUE, token = TRUE)
count <- 0L
while (TRUE) {
  count <- count + 1L
  out <- daemons()$daemons
  if (!is.matrix(out)) {
    break;
  }
  msleep(10) # or msleep(1) or msleep(100)
}
```

Could be related, but I am seeing a lot of `"'errorValue' int 7 | Object closed"` in the [transient throughput worker test](https://github.com/wlandau/crew/blob/61/tests/throughput/test-transient.R). I will try to create a reproducible example without using `crew`.
